mongo_orchestration/sharded_clusters.py

```diff
@@ -164,9 +155,10 @@ def only_x509(config):
                     client = ReplicaSets()._storage[instance_id].connection()
                 db = client[self.auth_source]
                 if self.x509_extra_user:
-                    db.add_user(DEFAULT_SUBJECT, roles=self._user_roles(client))
-                if self.login:
-                    db.add_user(**secondary_login)
+                    db.add_user(DEFAULT_SUBJECT, roles=roles)
+
+                create_user(db, mongos_version, self.login, self.password,
```

Comment:
We don't need the `self.login` check anymore?

Reply:
Nope, this section of code is already under an `if self.login:` branch so the condition was always true.